02-react-week-1/intro-to-state.md

```diff
@@ -0,0 +1,124 @@
+
+Here are a couple of quick examples of state:
+
+- The details of a Driver in a ride share app, including whether or not they are available to drive right now.
+- The moment to moment position of Pac-Man and the Ghosts during a game of Pac-Man.
+- The current time on a stopwatch or clock.
+- The restaurant that is currently selected while looking at Google Maps.
+
+Importantly, all of these things can be tracked as data, and they are all things that are likely to change while the program is running. Put another way, 'state' is a way of naming data in our program that we expect to change!
+
+So far, our React apps have had quite a bit of data, but it hasn't really had the opportunity to change much! In order to show a simple example, let's take a look at a brand new kind of component: the class-based, or "classical" component.
+
+## Class-Based Components
+
+There are two main types of components in React.  _Class Components_ and _Functional Components_.  Until now all the components you created in React were functions like this:
+
+```javascript
+// components/Student.js
+
+//Our cute functional components that we know and love.
+const Student = (props) => {
+  return (
+    <div>
+      <h3>{props.fullName}</h3>
+      <p>{props.email}</p>
+    </div>
+  );
+};
+```
+
+These components are displaying our data right now, but they aren't able to update or adjust that data! Because there's no way to update the data given to them, they can be called "stateless components". But what if we wanted a straightforward way to track state in a component? What existing tool combines the behavior of functions with the storage ability of variables? That's right! Classes!
+
+Let's take a quick look at what the functional component above would look like if it was class-based instead:
+
+```javascript
+// components/Student.js
+import React from 'react';
+
+class Student extends React.Component {
+  constructor(props) {
+    super(props); //props is passed in to the constructor of the component...
+    this.state = {
+      fullname: this.props.fullname,
+    }
+  }
+  
+  //It's okay to not understand what's going on here yet. 
+  //Long story short, I can find out the contents of my <input> 
+  //and send them back to the component's state!
+  onNameChange = () => {
+    this.setState({fullname: event.target.value})
+  }
+
+  render() {
+
+    //...and when we want to access that data, we can use `this`
+    return (
+      <div>
+        <h3>{this.state.fullname}</h3>
+        <p>{this.props.email}</p>
+        Change the Name: <input onChange={this.onNameChange}></input>
+      </div>
+    );
+  }
+}
+
+export default Student;
+```
+
+And if you want to play around with this live, [check out this codepen.](https://codepen.io/dhelmgren/pen/dypRybV) Don't worry about how the pieces we haven't talked about yet work, just try it out and confirm that we can change the name of the student.
+
```

Comment:
The codepen example will need to be updated with the new version of the code (present button rather than name example)

Reply:
I'll get started on that.